### PR TITLE
Added empty service

### PR DESCRIPTION
**Pull request by Vasis3038**
Added empty service
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1127

### CODE REVIEW

File: src/SurveyService/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        .Services.BuildServiceProvider();
+
+      return builder
+        .GetRequiredService<IOptions<MvcOptions>>()
+        .Value
+        .InputFormatters
+        .OfType<NewtonsoftJsonPatchInputFormatter>()
+        .First();
+    }
+
```
Comment (VLiamin) [inline src/SurveyService/Startup.cs]: empty lines

**Review by AlfaRedDog — APPROVED** [review]

**Review by maybee16 — APPROVED** [review]

File: src/SurveyService/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+namespace LT.DigitalOffice.SurveyService
+{
+  public class Startup : BaseApiInfo
+  {
+    private readonly BaseServiceInfoConfig _serviceInfoConfig;
+
+    private readonly RabbitMqConfig _rabbitMqConfig;
+
+    public const string CorsPolicyName = "LtDoCorsPolicy";
+
```
Comment (VLiamin) [inline src/SurveyService/Startup.cs:33]: empty line

File: src/SurveyService/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+      services.AddMassTransitHostedService();
+    }
+
+    private void ConfigureConsumers(IServiceCollectionBusConfigurator x)
+    {
+    }
+
+    private void ConfigureEndpoints(
+        IBusRegistrationContext context,
```
Comment (VLiamin) [inline src/SurveyService/Startup.cs]: spaces

File: src/SurveyService/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+      _serviceInfoConfig = Configuration
+        .GetSection(BaseServiceInfoConfig.SectionName)
+        .Get<BaseServiceInfoConfig>();
+
+      _rabbitMqConfig = Configuration
+        .GetSection(BaseRabbitMqConfig.SectionName)
+        .Get<RabbitMqConfig>();
+
+      Version = "1.0.0.0";
+      Description = "SurveyService is an API that intended to work with surveies.";
```
Comment (VLiamin) [inline src/SurveyService/Startup.cs]: May be surveys, instead of surveies? https://englishmore.ru/dictionary/mnozhestvennoe-chislo-survey

**Review by Levliv — APPROVED** [review]

File: Common.Configurations.targets [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    <Optimize>true</Optimize>
+    <DefineConstants>TRACE;RELEASE</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(AssemblyName.Contains('.UnitTests'))">
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup Condition="($(AssemblyName.Contains('.Provider')) Or $(AssemblyName.Contains('.Models.Db'))) And $(AssemblyName.Contains('.UnitTests')) != 'true'">
+  </ItemGroup>
```
Comment (vorobevamv) [inline Common.Configurations.targets]: lines 28+29 - don't need it, remove

File: Common.Configurations.targets [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+  <ItemGroup Condition="$(AssemblyName.Contains('.UnitTests')) != 'true'">
+	  <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.63" />
+	  <PackageReference Include="LT.DigitalOffice.Kernel.BrokerSupport" Version="1.0.7" />
+	  <PackageReference Include="LT.DigitalOffice.Kernel.EFSupport" Version="1.0.1" />
+	  <PackageReference Include="DigitalOffice.Models" Version="1.0.1" />
+  </ItemGroup>
```
Comment (vorobevamv) [inline Common.Configurations.targets:33]: lines 32-35 -  fix tabs

File: src/SurveyService.Broker/SurveyService.Broker.csproj [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+	<Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
+
+	<ItemGroup>
+		<PackageReference Include="Microsoft.AspNetCore.Mvc.Core" Version="2.2.5" />
```
Comment (vorobevamv) [inline src/SurveyService.Broker/SurveyService.Broker.csproj]: fix tab

File: src/SurveyService.Broker/SurveyService.Broker.csproj [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+	</ItemGroup>
+
+	<ItemGroup>
+		<ProjectReference Include="..\SurveyService.Data\SurveyService.Data.csproj" />
+		<ProjectReference Include="..\SurveyService.Mappers\SurveyService.Mappers.csproj" />
+		<ProjectReference Include="..\SurveyService.Models.Db\SurveyService.Models.Db.csproj" />
```
Comment (vorobevamv) [inline src/SurveyService.Broker/SurveyService.Broker.csproj]: 10-12 - fix tabs

File: src/SurveyService.Data.Provider.MsSql.Ef/SurveyService.Data.Provider.MsSql.Ef.csproj [hunk 9]
```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+	<Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
+
+	<ItemGroup>
+		<ProjectReference Include="..\SurveyService.Data.Provider\SurveyService.Data.Provider.csproj" />
+		<ProjectReference Include="..\SurveyService.Models.Dto\SurveyService.Models.Dto.csproj" />
```
Comment (vorobevamv) [inline src/SurveyService.Data.Provider.MsSql.Ef/SurveyService.Data.Provider.MsSql.Ef.csproj]: 6+7 - fix tabs

File: src/SurveyService.Data.Provider/SurveyService.Data.Provider.csproj [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+	<Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
+
+	<ItemGroup>
+		<ProjectReference Include="..\SurveyService.Models.Db\SurveyService.Models.Db.csproj" />
```
Comment (vorobevamv) [inline src/SurveyService.Data.Provider/SurveyService.Data.Provider.csproj]: fix tab

File: src/SurveyService.Data/SurveyService.Data.csproj [hunk 11]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+	<Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
+
+	<ItemGroup>
+		<ProjectReference Include="..\SurveyService.Data.Provider.MsSql.Ef\SurveyService.Data.Provider.MsSql.Ef.csproj" />
+		<ProjectReference Include="..\SurveyService.Models.Db\SurveyService.Models.Db.csproj" />
+		<ProjectReference Include="..\SurveyService.Models.Dto\SurveyService.Models.Dto.csproj" />
+	</ItemGroup>
```
Comment (vorobevamv) [inline src/SurveyService.Data/SurveyService.Data.csproj]: 6-8 -  fix tabs

File: src/SurveyService.Mappers/SurveyService.Mappers.csproj [hunk 12]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+	<Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
+
+	<ItemGroup>
+		<ProjectReference Include="..\SurveyService.Models.Db\SurveyService.Models.Db.csproj" />
+		<ProjectReference Include="..\SurveyService.Models.Dto\SurveyService.Models.Dto.csproj" />
+	</ItemGroup>
```
Comment (vorobevamv) [inline src/SurveyService.Mappers/SurveyService.Mappers.csproj]: 6+7 - fix tabs

File: src/SurveyService.Models.Dto/SurveyService.Models.Dto.csproj [hunk 13]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+	<Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
+
+	<ItemGroup>
+		<PackageReference Include="Microsoft.AspNetCore.Mvc.Core" Version="2.2.5" />
```
Comment (vorobevamv) [inline src/SurveyService.Models.Dto/SurveyService.Models.Dto.csproj]: fix tab

File: src/SurveyService.Validation/SurveyService.Validation.csproj [hunk 14]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+	<Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
+
+	<ItemGroup>
+		<ProjectReference Include="..\SurveyService.Data\SurveyService.Data.csproj" />
+		<ProjectReference Include="..\SurveyService.Models.Dto\SurveyService.Models.Dto.csproj" />
```
Comment (vorobevamv) [inline src/SurveyService.Validation/SurveyService.Validation.csproj]: 6+7 - fix tabs

File: src/SurveyService/Startup.cs [hunk 15]
```diff
@@ -0,0 +1,206 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Serialization;
+
+namespace LT.DigitalOffice.SurveyService
+{
+  public class Startup : BaseApiInfo
+  {
+    private readonly BaseServiceInfoConfig _serviceInfoConfig;
+
```
Comment (vorobevamv) [inline src/SurveyService/Startup.cs]: remove empty line: 35+37 are both private readonly, don't need to separate them

File: src/SurveyService/Startup.cs [hunk 16]
```diff
@@ -0,0 +1,206 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Serialization;
+
+namespace LT.DigitalOffice.SurveyService
```
Comment (vorobevamv) [inline src/SurveyService/Startup.cs:25]: please, fix order in Startup class:
1. constructor
2. public methods
3. private methods

File: src/SurveyService/Startup.cs [hunk 17]
```diff
@@ -0,0 +1,206 @@
...
+    private readonly RabbitMqConfig _rabbitMqConfig;
+
+    public const string CorsPolicyName = "LtDoCorsPolicy";
+
+
+    public IConfiguration Configuration { get; }
+
+    #region private methods
+
+    private static NewtonsoftJsonPatchInputFormatter GetJsonPatchInputFormatter()
```
Comment (vorobevamv) [inline src/SurveyService/Startup.cs]: don't need this method for now

File: src/SurveyService/Startup.cs [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+            cfg.Host(_rabbitMqConfig.Host, "/", host =>
+              {
+                host.Username(username);
+                host.Password(password);
+              });
+
+            ConfigureEndpoints(context, cfg, _rabbitMqConfig);
+          });
+
+        ConfigureConsumers(busConfigurator);
```
Comment (vorobevamv) [inline src/SurveyService/Startup.cs]: remove

File: src/SurveyService/Startup.cs [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+      {
+        busConfigurator.UsingRabbitMq((context, cfg) =>
+          {
+            cfg.Host(_rabbitMqConfig.Host, "/", host =>
+              {
+                host.Username(username);
+                host.Password(password);
+              });
+
+            ConfigureEndpoints(context, cfg, _rabbitMqConfig);
```
Comment (vorobevamv) [inline src/SurveyService/Startup.cs]: remove

File: src/SurveyService/Startup.cs [hunk 20]
```diff
@@ -0,0 +1,206 @@
...
+
+        ConfigureConsumers(busConfigurator);
+
+        busConfigurator.AddRequestClients(_rabbitMqConfig);
+      });
+
+      services.AddMassTransitHostedService();
+    }
+
+    private void ConfigureConsumers(IServiceCollectionBusConfigurator x)
```
Comment (vorobevamv) [inline src/SurveyService/Startup.cs]: do you need consumers in service?

File: src/SurveyService/Startup.cs [hunk 21]
```diff
@@ -0,0 +1,206 @@
...
+    {
+    }
+
+    private void ConfigureEndpoints(
```
Comment (vorobevamv) [inline src/SurveyService/Startup.cs]: this method is used for setting endpoints in other services, which receive information from this service;
will you send data from SurveyService to other services?

File: src/SurveyService/Startup.cs [hunk 22]
```diff
@@ -0,0 +1,206 @@
...
+        .GetSection(BaseRabbitMqConfig.SectionName)
+        .Get<RabbitMqConfig>();
+
+      Version = "1.0.0.0";
+      Description = "SurveyService is an API that intended to work with surveies.";
+      StartTime = DateTime.UtcNow;
+      ApiName = $"LT Digital Office - {_serviceInfoConfig.Name}";
+    }
+
+
```
Comment (vorobevamv) [inline src/SurveyService/Startup.cs]: remove empty line

File: src/SurveyService/Startup.cs [hunk 23]
```diff
@@ -0,0 +1,206 @@
...
+      services
+        .AddControllers()
+        .AddJsonOptions(options =>
+        {
+          options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
+        })
+        .AddNewtonsoftJson();
+    }
+
+
```
Comment (vorobevamv) [inline src/SurveyService/Startup.cs]: remove empty line: for separating methods use one empty line (not two)

File: src/SurveyService/Startup.cs [hunk 24]
```diff
@@ -0,0 +1,206 @@
...
+          {
+            { HealthStatus.Unhealthy, 200 },
+            { HealthStatus.Healthy, 200 },
+            { HealthStatus.Degraded, 200 },
+          },
+          Predicate = check => check.Name != "masstransit-bus",
+          ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
+        });
+      });
+
```
Comment (vorobevamv) [inline src/SurveyService/Startup.cs:132]: remove empty line

File: src/SurveyService/SurveyService.csproj [hunk 25]
```diff
@@ -0,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+	
+	<Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
+
+	<ItemGroup>
+		<ProjectReference Include="..\SurveyService.Broker\SurveyService.Broker.csproj" />
+		<ProjectReference Include="..\SurveyService.Business\SurveyService.Business.csproj" />
+		<ProjectReference Include="..\SurveyService.Validation\SurveyService.Validation.csproj" />
+	</ItemGroup>
```
Comment (vorobevamv) [inline src/SurveyService/SurveyService.csproj]: 6-8 - fix tabs

File: src/SurveyService/SurveyService.csproj [hunk 26]
```diff
@@ -0,0 +1,24 @@
...
+		<PackageReference Include="AspNetCore.HealthChecks.UI.Client" Version="6.0.5" />
+		<PackageReference Include="FluentValidation.AspNetCore" Version="11.2.1" />
+		<PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="6.0.8" />
+		<PackageReference Include="Serilog.AspNetCore" Version="6.0.1" />
+		<PackageReference Include="Serilog.Enrichers.Environment" Version="2.2.0" />
+		<PackageReference Include="Serilog.Enrichers.Process" Version="2.0.2" />
+		<PackageReference Include="Serilog.Enrichers.Thread" Version="3.1.0" />
+		<PackageReference Include="Serilog.Settings.Configuration" Version="3.3.0" />
+		<PackageReference Include="Serilog.Sinks.Seq" Version="5.1.1" />
+	</ItemGroup>
```
Comment (vorobevamv) [inline src/SurveyService/SurveyService.csproj]: 12-21 fix tabs

File: src/SurveyService/appsettings.Production.json [hunk 27]
```diff
@@ -0,0 +1,10 @@
+{
+  "Serilog": {
+  },
+  "ConnectionStrings": {
+    "SQLConnectionString": "Server=127.0.0.1,20340;User=sa;Password=User_1234;Database=SurveyDB;"
```
Comment (vorobevamv) [inline src/SurveyService/appsettings.Production.json]: Password=Survey_1234;

**Comment by vorobevamv** [comment]
please, add in PR description reference to task in Redmine

File: src/SurveyService.Business/SurveyService.Business.csproj [hunk 28]
```diff
@@ -0,0 +1,16 @@
...
+  <ItemGroup>
+    <ProjectReference Include="..\SurveyService.Broker\SurveyService.Broker.csproj" />
+    <ProjectReference Include="..\SurveyService.Data\SurveyService.Data.csproj" />
+    <ProjectReference Include="..\SurveyService.Mappers\SurveyService.Mappers.csproj" />
+    <ProjectReference Include="..\SurveyService.Validation\SurveyService.Validation.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="Commands\" />
+  </ItemGroup>
```
[the remark this answers is not in the document]
Comment (Vasis3038) [reply src/SurveyService.Business/SurveyService.Business.csproj:14]: then folder will disappear from vs, is it ok?

File: src/SurveyService/Startup.cs (hunk 21 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: I guess no, I'll remove this method

File: src/SurveyService/Startup.cs (hunk 20 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: I will need to get info from UserService, so I guess yes..

File: src/SurveyService/Startup.cs (hunk 2 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs:33]: done

File: src/SurveyService/Startup.cs (hunk 1 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: done

File: src/SurveyService/Startup.cs (hunk 3 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: done

File: src/SurveyService/Startup.cs (hunk 4 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: done

File: Common.Configurations.targets (hunk 5 above)
Comment (Vasis3038) [reply Common.Configurations.targets]: done

File: src/SurveyService.Broker/SurveyService.Broker.csproj (hunk 7 above)
Comment (Vasis3038) [reply src/SurveyService.Broker/SurveyService.Broker.csproj]: done

File: src/SurveyService.Data.Provider.MsSql.Ef/SurveyService.Data.Provider.MsSql.Ef.csproj (hunk 9 above)
Comment (Vasis3038) [reply src/SurveyService.Data.Provider.MsSql.Ef/SurveyService.Data.Provider.MsSql.Ef.csproj]: done

File: src/SurveyService.Data.Provider/SurveyService.Data.Provider.csproj (hunk 10 above)
Comment (Vasis3038) [reply src/SurveyService.Data.Provider/SurveyService.Data.Provider.csproj]: done

File: src/SurveyService.Mappers/SurveyService.Mappers.csproj (hunk 12 above)
Comment (Vasis3038) [reply src/SurveyService.Mappers/SurveyService.Mappers.csproj]: done

File: src/SurveyService.Models.Dto/SurveyService.Models.Dto.csproj (hunk 13 above)
Comment (Vasis3038) [reply src/SurveyService.Models.Dto/SurveyService.Models.Dto.csproj]: done

File: src/SurveyService.Validation/SurveyService.Validation.csproj (hunk 14 above)
Comment (Vasis3038) [reply src/SurveyService.Validation/SurveyService.Validation.csproj]: done

File: src/SurveyService/Startup.cs (hunk 15 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: done

File: src/SurveyService/Startup.cs (hunk 17 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: done

File: src/SurveyService/Startup.cs (hunk 19 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: done

File: src/SurveyService/Startup.cs (hunk 22 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: done

File: src/SurveyService/Startup.cs (hunk 18 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: done

File: src/SurveyService/Startup.cs (hunk 16 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs:25]: done

File: src/SurveyService/Startup.cs (hunk 21 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: done

File: src/SurveyService/Startup.cs (hunk 24 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs:132]: done

File: src/SurveyService/SurveyService.csproj (hunk 25 above)
Comment (Vasis3038) [reply src/SurveyService/SurveyService.csproj]: done

File: src/SurveyService/appsettings.Production.json (hunk 27 above)
Comment (Vasis3038) [reply src/SurveyService/appsettings.Production.json]: done

File: src/SurveyService.Data/SurveyService.Data.csproj (hunk 11 above)
Comment (Vasis3038) [reply src/SurveyService.Data/SurveyService.Data.csproj]: done

File: src/SurveyService/Startup.cs (hunk 20 above)
Comment (vorobevamv) [reply src/SurveyService/Startup.cs]: Consumers are in services, which receive requests from other services, goes to DB and send response.
To get info from UserService you need consumer in UserService, not here)

File: src/SurveyService/SurveyService.csproj (hunk 26 above)
Comment (Vasis3038) [reply src/SurveyService/SurveyService.csproj]: done

File: src/SurveyService.Broker/SurveyService.Broker.csproj (hunk 8 above)
Comment (Vasis3038) [reply src/SurveyService.Broker/SurveyService.Broker.csproj]: done

File: Common.Configurations.targets (hunk 6 above)
Comment (Vasis3038) [reply Common.Configurations.targets:33]: done

File: src/SurveyService/Startup.cs (hunk 20 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: Got it
done

File: src/SurveyService/Startup.cs (hunk 23 above)
Comment (Vasis3038) [reply src/SurveyService/Startup.cs]: done

**Review by vorobevamv — APPROVED** [review]

**Pull request merged**